python/pyspark/rdd.py

```diff
@@ -869,7 +871,10 @@ def collect(self):
             to be small, as all the data is loaded into the driver's memory.
         """
         with SCCallSiteSync(self.context) as css:
-            sock_info = self.ctx._jvm.PythonRDD.collectAndServe(self._jrdd.rdd())
+            java_map = MapConverter().convert(self.context.getLocalProperties(),
+                                              self.context._gateway._gateway_client)
+            sock_info = self.ctx._jvm.PythonRDD.collectAndServe(
+                self._jrdd.rdd() ,java_map)
```

Comment:
Would that work if we use UDF + count and use TaskContext's local property access?